Add Toolbar (#6358)
jpmorganchase/salt-ds: site/docs/components/toolbar/accessibility.mdx

---
title:
  $ref: ./#/title
layout: DetailComponent
sidebar:
  exclude: true
data:
  $ref: ./#/data
---

## Best practices

> - Avoid controls that utilize arrow keys, like a text input, in the start or middle of a toolbar. These controls disrupt arrow-key item navigation and make Tab move to the next item instead of exiting, breaking expected toolbar behavior.

Reviewer: ```suggestion
- Avoid controls that utilize arrow keys, like a text input, in the start or middle of a toolbar. These controls disrupt arrow-key navigation and make Tab move to the next item instead of exiting, breaking expected toolbar behavior.
```

>   - If unavoidable, include only one and place it last to preserve arrow-key navigation and normal Tab exit.

Reviewer: ```suggestion
  - If unavoidable, include only one and place it last in the toolbar to preserve arrow-key navigation and normal Tab exit.
```

- Use an `aria-label` when there is no visible label on the toolbar element and `aria-labelledby` when there is a visible label.

> - With transparent toolbar, avoid spanning the full width of a page unless it contains enough controls to be recognized as a toolbar. Otherwise, use a bordered toolbar.

Reviewer: ```suggestion
- With transparent toolbar, ensure close spacing between controls and groups of controls so it's clearly recognizable as a toolbar. Otherwise, use a bordered toolbar to reinforce the toolbar container.
```
@jake-costa thoughts on this callout for transparent toolbar?

## Keyboard interactions

<KeyboardControls>

> <KeyboardControl keyOrCombos="Tab">

Reviewer 2: Not sure if its worth mentioning focus moving to last focused element in the toolbar
